src/libraries/Native/Unix/System.Native/pal_signal.c

```diff
@@ -309,21 +366,68 @@ int32_t InitializeSignalHandlingCore()
     }
 
     // Finally, register our signal handlers
-    // We don't handle ignored SIGINT/SIGQUIT signals. If we'd setup a handler, our child
-    // processes would reset to the default on exec causing them to terminate on these signals.
+    // We don't handle ignored SIGINT/SIGQUIT/SIGHUP/SIGTERM signals. If we'd
+    // setup a handler, our child processes would reset to the default on exec
+    // causing them to terminate on these signals.
     InstallSignalHandler(SIGINT , /* skipWhenSigIgn */ true);
     InstallSignalHandler(SIGQUIT, /* skipWhenSigIgn */ true);
+    InstallSignalHandler(SIGHUP, /* skipWhenSigIgn */ true);
+    InstallSignalHandler(SIGTERM, /* skipWhenSigIgn */ true);
+
     InstallSignalHandler(SIGCONT, /* skipWhenSigIgn */ false);
     InstallSignalHandler(SIGCHLD, /* skipWhenSigIgn */ false);
     InstallSignalHandler(SIGWINCH, /* skipWhenSigIgn */ false);
 
     return 1;
 }
 
+int32_t SystemNative_RegisterForPosixSignal(PosixSignal signal, PosixSignalHandler signalHandler)
+{
+    assert(signalHandler != NULL);
+    assert(g_posixSignalHandler == NULL || g_posixSignalHandler == signalHandler);
+
+    if (signal >= 0 || -signal > PosixSignalCount)
```

Comment:
Am I reading this right? Does this mean there will be no way to register handlers for signals that aren't preapproved by the runtime?

If so, this seems like a big downside. For one thing, this would stop Mono.Unix from ever using this API (instead of its native library helpers) since it supports the full set of signals that POSIX supports, including the real time signal range.

Reply:
https://github.com/dotnet/runtime/issues/50527#issuecomment-825858901 didn't mention this.
I see the comments below describe this behavior, so I'll look into it.

Reply:
Ah, I had missed the API review comment.

But various comments from @davidfowl and @bartonjs do seem to indicate this aspect is still intended:

* https://github.com/dotnet/runtime/issues/50527#issuecomment-813207366
* https://github.com/dotnet/runtime/issues/50527#issuecomment-825879127
* https://github.com/dotnet/runtime/issues/50527#issuecomment-825911645